src/main/java/com/empowerops/linqalike/IList.java

```diff
@@ -0,0 +1,167 @@
+package com.empowerops.linqalike;
+
+import com.empowerops.linqalike.queries.FastSize;
+import org.pcollections.PVector;
+
+import javax.annotation.Nonnull;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+/**
+ * Created by Geoff on 2015-12-02.
+ */
+@SuppressWarnings("deprecation") //intentional and carried forward.
+public class IList<T> implements ImmutableCollection<T>, List<T>, DefaultedQueryable<T>, QueryableList<T>, FastSize{
+
+    private static final IList Empty = new IList();
+    @SuppressWarnings("unchecked") //thanks to immutableness this is safe!
+    public static <T> IList<T> empty(){ return Empty; }
+
+
+    private final PVector<T> backingList;
+
+    public IList(){
+        backingList = org.pcollections.Empty.vector();
+    }
+
+    public IList(PVector<T> backingList){
+        this.backingList = backingList;
+    }
+
+    @SafeVarargs
+    public IList(T... initialElements){
+        PVector<T> backingList = org.pcollections.Empty.vector();
+        for(T initial : initialElements){
+            backingList = backingList.plus(initial);
+        }
+        this.backingList = backingList;
+    }
+
+    @Override
+    public int size() {
+        return backingList.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return backingList.isEmpty();
+    }
+
+    @Override
+    public boolean contains(Object o) {
+        return backingList.contains(o);
+    }
+
+    @Override
+    public int indexOf(Object o) {
+        return backingList.indexOf(o);
+    }
+
+    @Override
+    public int lastIndexOf(Object o) {
+        return backingList.lastIndexOf(o);
+    }
+
+    @Override
+    public @Nonnull ListIterator<T> listIterator() {
+        return backingList.listIterator();
+    }
+
+    @Override
+    public @Nonnull ListIterator<T> listIterator(int index) {
+        return backingList.listIterator(index);
+    }
+
+    @Override
+    public @Nonnull IList<T> subList(int fromIndex, int toIndex) {
+        return new IList<>(backingList.subList(fromIndex, toIndex));
+    }
+
+    @Override
+    public T get(int index) {
+        return backingList.get(index);
+    }
+
+    @Override
+    public @Nonnull Iterator<T> iterator() {
+        return listIterator();
+    }
+
+    @Override
+    public @Nonnull Object[] toArray() {
+        return backingList.toArray();
+    }
+
+    @Override
+    public @Nonnull <T1> T1[] toArray(T1[] a) {
+        return backingList.toArray(a);
+    }
+
+    @Override
+    public IList<T> with(T toInclude) {
+        return new IList<>(backingList.plus(toInclude));
+    }
+
+    public IList<T> with(int index, T toInclude){
+        //TODO hmm, this means my 'with' is their 'add' and their 'with' is juc's 'set' method.
```

Comment:
I really dont like 

```Java
list.plus(member)
```

I like _with_, but if it causes confusion I'd settle for `concat`.